rpc/swap.go

```diff
@@ -44,22 +46,44 @@ func NewSwapService(
 	}
 }
 
+// PastSwapInfo contains a swap ID and its start and end time.
+type PastSwapInfo struct {
+	ID        string    `json:"id"`
+	StartTime time.Time `json:"startTime"`
+	EndTime   time.Time `json:"endTime"`
```

Comment:
It was on my TODO list to start adding validation tags on these methods. They are a lower priority, because they are not peer-to-peer messages. Would it be reasonable to add `validate:"required"` to all 3 of these fields now? It provides code documentation. If I see the tag on `startTime` and not on `endTime`, as an example, I'll know instantly that the `endTime` field might not always be present.

Reply:
yeah, it would be reasonable to mark these all as required, do you want me to do that here or will you do it in another PR?

Reply:
Let's start doing it anywhere we can now, especially if we are modifying a serialized structure or creating a new one. It will make for less work and code review when I add it to the others.